Sources/PostgresNIO/Connection/PostgresConnection.swift

```diff
@@ -378,6 +378,28 @@ extension PostgresConnection {
         ).get()
     }
 
+    /// Creates a new connection to a Postgres server.
+    ///
+    /// This method uses an `EventLoop` from `MultiThreadedEventLoopGroup.singleton`, which is provided by SwiftNIO
+    ///
+    /// - Parameters:
+    ///   - configuration: A ``Configuration`` that shall be used for the connection
+    ///   - connectionID: An `Int` id, used for metadata logging
+    ///   - logger: A logger to log background events into
+    /// - Returns: An established  ``PostgresConnection`` asynchronously that can be used to run queries.
+    public static func connect(
+        configuration: PostgresConnection.Configuration,
```

Comment:
We're still figuring out the patterns here so I don't know if this is a good suggestion but maybe a defaulted parameter `on eventLoop: EventLoop = MultiThreadedEventLoopGroup.singleton.any()` could work?

Reply:
Not sure about this in the context of NIOTS... If we can import NIOTS, I would love to use the NIOTS EventLoop in the future for those cases. Defaulting inside the API might prove problematic then.

Reply:
Once NIOTS is supported, I'd suggest to go down the route of my AHC PR: https://github.com/swift-server/async-http-client/pull/697/files#diff-914333008acc7684e7d98c700381a134ecdf49f095ce644e5898e857a16feabfR880

Then PostgresNIO could provide a `PostgresNIO.defaultEventLoopGroup` which would return either `MTELG.singleton` or `NIOTSELG.singleton`. And apart from the default value nothing needs to change.

If you don't want to "leak" that we're using MTELG.singleton today, then I'd suggest to add the `static var defaultEventLoopGroup` to PostgresNIO with this PR.

Reply:
@weissi Lovely! @tkrajacic can you take ti from here?

Reply:
@tkrajacic Just discussed this a bit further with @weissi:

```swift
extension PostgresConnection {
     /// Returns the default `EventLoopGroup` singleton, automatically selecting the best for the platform.
     ///
     /// This will select the concrete `EventLoopGroup` depending which platform this is running on.
     public static var defaultEventLoopGroup: EventLoopGroup {
 #if canImport(Network)
         if #available(OSX 10.14, iOS 12.0, tvOS 12.0, watchOS 6.0, *) {
             return NIOTSEventLoopGroup.singleton
         } else {
             return MultiThreadedEventLoopGroup.singleton
         }
 #else
         return MultiThreadedEventLoopGroup.singleton
 #endif
     }
 }
```

And then we should default the existing connect method to `. defaultEventLoopGroup`, instead of creating a separate connect method.

Reply:
@fabianfett I have all that locally staged, but I was waiting until I could use the new NIOTS default eventloop group. So I am waiting for that to get released